src/vs/workbench/api/browser/mainThreadTask.ts

```diff
@@ -495,6 +493,7 @@ export class MainThreadTask implements MainThreadTaskShape {
 					dto.name = ((dto.name === undefined) ? '' : dto.name); // Using an empty name causes the name to default to the one given by the provider.
 					return Promise.resolve(this._proxy.$resolveTask(handle, dto)).then(resolvedTask => {
 						if (resolvedTask) {
+							resolvedTask.icon = task.configurationProperties.icon;
 							return TaskDTO.to(resolvedTask, this._workspaceContextServer, true);
```

Comment:
Can we move this into `TaskDTO.to` and pass in either `task.configurationProperties` or `task.configurationProperties.icon` to it? That way we don't need to touch the TaskDTO which is designed exclusively for transferring the data between processes.

Reply:
`TaskDTO.to` is called all of the place from contexts that don't have access to `task.configurationProperties`

Reply:
tried that and it should work but still isn't 🤷🏼‍♀️

Reply:
pushed